### PR TITLE
feat added dark mode

### LINKED ISSUE #3: Add Dark Mode
Introduce a dark mode option to provide users with an alternative color scheme that is easier on the eyes and conserves battery life on OLED screens.

### PR DESCRIPTION
**Pull request by YuvrajSingh3110**
## Description
Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.
Added a toggle on the profile page to change theme of the app. Used shared preferences to store locally the current theme mode.

Fixes #3 

Mention issue number in the above issue bracket.

## Type of change
Please delete options that are not relevant.

- [✔ ] New feature (non-breaking change which adds functionality)

## Checklist:
- [ ✔] My code follows the style guidelines of this project
- [ ✔] I have performed a self-review of my own code
- [ ✔] I have commented my code, particularly in hard-to-understand areas
- [ ✔] I have made corresponding changes to the documentation
- [ ✔] My changes generate no new warnings
- [ ✔] I have added tests that prove my fix is effective or that my feature works
- [ ✔] New and existing unit tests pass locally with my changes
- [ ✔] Any dependent changes have been merged and published in downstream modules

## Screenshots (if appropriate):
image

## Additional Notes:
Include any additional information that you think is important for the reviewer to know.
As the profile, login and signup screens use an image background so dark mode could not implemented on them and the splash screen cannot have black background as the logo shown is itself black

### CODE REVIEW

File: pubspec.yaml
Comment (Photon3009) [inline pubspec.yaml:1]: undo this unnecessary change

**Review by Photon3009 — CHANGES_REQUESTED** [review]
When on the login screen in the light theme, the text is not visible because it is white in color. Also, in dark mode, the login and sign-up screens do not look good. Therefore, let's avoid implementing theme changes on the login and sign-up screens. Additionally, please add a toast message when the user switches between themes, as the profile screen does not change. It would look good if users receive a toast message.

Please make these changes and thoroughly test your code in various corner edges. You've done an amazing work so far!

**Comment by YuvrajSingh3110** [comment]
Please review the changes @Photon3009

**Review by Photon3009 — APPROVED** [review]

**Comment by Photon3009** [comment]
good work @YuvrajSingh3110
